adam/curriculum/organize_objects.py

```diff
@@ -0,0 +1,128 @@
+
+        if not concept_path.is_dir():
+            logger.warning(
+                "Concept %s not found in given directory", object_concept.handle
+            )
+            continue
+
+        for img_path in sorted(concept_path.glob("*.png")):
+            yield object_concept.handle, img_path
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Utility script to reorganize object curricula"
+    )
+    parser.add_argument("input_img_dir", type=Path, help="Input set of images.")
+    parser.add_argument(
+        "base_curriculum_dir", type=Path, help="Base directory for curriculum storage."
+    )
+    parser.add_argument(
+        "curriculum_name", type=str, help="The name of the curriculum to create."
+    )
+    parser.add_argument(
+        "--train-num",
+        type=int,
+        default=10,
+        help="The number of samples to put in the training curriculum. The rest of the samples will go the test curriculum.",
+    )
+    parser.add_argument(
+        "--overwrite",
+        action="store_true",
+        help="Enable writing over an existing curriculum directory.",
+    )
+    parser.add_argument(
+        "--image-size",
+        nargs=2,
+        metavar=("img_width", "img_height"),
+        help="The values to resize the give curriculum images to. Must be 16x9 & both divisible by 8.",
+        default=(384, 216),
+        type=int,
+    )
+    args = parser.parse_args()
+
+    source_img_dir: Path = args.input_img_dir
+    train_curriculum_dir: Path = args.base_curriculum_dir / "train" / args.curriculum_name
+    test_curriculum_dir: Path = (
+        args.base_curriculum_dir / "test" / f"{args.curriculum_name}_eval"
+    )
+
+    img_width = args.image_size[0]
+    img_height = args.image_size[1]
+
+    assert img_width % 16 == 0
+    assert img_height % 8 == 0 and img_height % 9 == 0
+
+    train_curriculum_dir.mkdir(exist_ok=args.overwrite, parents=True)
+    test_curriculum_dir.mkdir(exist_ok=args.overwrite, parents=True)
+
+    train_scene_count = 0
+    test_scene_count = 0
+    img2 = None
+    object_names_to_count: Dict[str, int] = {}
+    for object_name, scene_img in img_iter(source_img_dir):
+        if object_names_to_count.get(object_name, 0) < args.train_num:
+            sit_dir = train_curriculum_dir / f"situation_{train_scene_count}"
+            train_scene_count += 1
+        else:
+            sit_dir = test_curriculum_dir / f"situation_{test_scene_count}"
+            test_scene_count += 1
+        sit_dir.mkdir(parents=True, exist_ok=True)
+
+        img = cv2.imread(str(scene_img))
+        img = cv2.resize(img, (img_width, img_height))
+        try:
+            img2 = cv2.cvtColor(img, cv2.COLOR_RGBA2RGB)
+            logger.debug(f"Image {scene_img} is RGBA")
+        except ValueError:
+            logger.debug(f"Image {scene_img} is RGB")
+            img2 = img
+        finally:
+            cv2.imwrite(str((sit_dir / "rgb_0.png")), img2)
```

Comment:
Redundant parentheses?

```suggestion
            cv2.imwrite(str(sit_dir / "rgb_0.png"), img2)
```